openapi/openapi.yaml

```diff
@@ -629,6 +629,19 @@ components:
           description: The type (or types) of descriptors available.
           items:
             $ref: "#/components/schemas/DescriptorType"
+        descriptor_type_version:
+          type: object
+          description: A map providing information about the language versions used in this tool. The keys should be the
+            same values in the `descriptor_type` table, and the value should be the actual language version for the given
+            `descriptor_type`
+          example: |
+            {
+              "WDL": "1.0",
+              "CWL": "v1.0.2",
+              "NFL": "22.04.0"
```

Comment:
I don't know Nextflow, but looking at a random NF-Core workflow registered on Dockstore, it [specifies a minimum Nextflow version](https://github.com/nf-core/rnaseq/blob/1.4.2/nextflow.config#L166). Then I found the Nextflow [docs on it](https://www.nextflow.io/docs/latest/config.html?highlight=nextflowversion#nextflow-version). 

I'm personally more used to the [NPM semantic versioning](https://docs.npmjs.com/about-semantic-versioning), 

In any case, it seems like this case should be covered one way or another.

Reply:
Hmmm. This seems a little different. 
The version of the Nextflow jar that a workflow is compatible with is different from the language version. 
Nextflow's equivalent of WDL draft-3, 1.0, etc. is DSL2 https://www.nextflow.io/docs/latest/dsl2.html

This does mean I would recommend adjusting the example though.

Reply:
Would be good to hear something about the relationship between workflow languages/types, workflow language/type versions and required workflow engine versions from the developers of workflow languages for which only a single engine exists. Particularly for Snakemake, I don't think there is such a thing as a workflow language version (but I may be wrong). That would help us to better map workflows to WES instances and increase the chance that a workflow will actually be processed successfully.

I suppose the most interesting question would probably be:
* Do Nextflow/Snakemake workflows require specific workflow engine versions?
* Do Nextflow/Snakemake workflows require specific workflow language versions? Is there a concept of language versions at all in Snakemake?
* Are Nextflow and Snakemake workflow engines strictly backwards compatible in their ability to process older workflows / workflow languages?

Perhaps @johanneskoester and @pditommaso would like to comment?